go/test/endtoend/cluster/mysqlctld_process.go

```diff
@@ -170,6 +175,22 @@ func (mysqlctld *MysqlctldProcess) IsHealthy() bool {
 }
 
 // HasShutdown checks if the process has been set to nil
-func (mysqlctld *MysqlctldProcess) HasShutdown() bool {
+func (mysqlctld *MysqlctldProcess) hasShutdown() bool {
 	return mysqlctld.process == nil
 }
+
+func (mysqlctld *MysqlctldProcess) WaitForMysqlctldShutdown(tablet *Vttablet) bool {
+	tmr := time.NewTimer(defaultOperationTimeout)
+	defer tmr.Stop()
+	for {
+		if tablet.MysqlctldProcess.hasShutdown() {
+			return true
+		}
+		select {
+		case <-tmr.C:
+			return false
+		default:
+		}
+		time.Sleep(defaultRetryDelay)
+	}
+}
```

Comment:
I don't think you need the primary tablet as input of this function, and you should use the receiver `*MysqlctldProcess` instead of `tablet.MysqlctldProcess` on line `186`

Reply:
ah good catch.. I moved this function under type during refactoring didn't realize it will no longer need the argument.